core/modules/layout/plugins/access/entity_id_layout_access.inc

```diff
@@ -33,6 +33,25 @@ class EntityIDLayoutAccess extends LayoutAccessNegatable {
     );
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  function formValidate($form, &$form_state) {
```

Comment:
I don't think we need any validation if we change the form element type to number.

Reply:
I guess 'number' needs to be added as a type to the documentation: https://docs.backdropcms.org/form_api. 

Maybe doing explicit validation is worth getting more explicit and helpful validation messages. Looking through `form_validate_number()`, it looks like the error message for entering a non-integral value (after appropriately setting `#step`) would be `'%name is not a valid number'`.

Also, in the PR I'm validating for the entity ID to exist. If we drop `formValidate()`, then we lose that bit of validation. Seems like it would be useful to have.